[CodeComplete] Complete argument labels after vararg

ahoppen opened the pull request:
Currently, if the code completion token is after a label that refers to a vararg, it is part of that `VarargExpansionExpr`, so we don’t suggest the subsequent parameter’s label.

Add special handling to detect this situation and suggest the parameter label.

Fixes rdar://76355192

ahoppen commented:
@swift-ci Please smoke test

ahoppen commented:
@swift-ci Please test

rintaro commented on test/IDE/complete_call_arg.swift:
```diff
@@ -866,3 +860,31 @@ func testClosurePlaceholderPrintsTypesOnlyIfNoInternalParameterNamesExist() {
+  }
+
+  private func test(value: Rdar76355192) {
+    value.test("test", xArg: #^COMPLETE_AFTER_VARARG^#)
+    // COMPLETE_AFTER_VARARG: Begin completions
+    // COMPLETE_AFTER_VARARG-DAG: Pattern/ExprSpecific:               {#yArg: Foo...#}[#Foo#];
+    // COMPLETE_AFTER_VARARG-DAG: Pattern/ExprSpecific:               {#zArg: Foo...#}[#Foo#];
+    // COMPLETE_AFTER_VARARG: End completions
+    value.test("test", xArg: #^COMPLETE_AFTER_VARARG_WITH_PREV_PARAM^#)
+    // COMPLETE_AFTER_VARARG_WITH_PREV_PARAM: Begin completions
+    // COMPLETE_AFTER_VARARG_WITH_PREV_PARAM-DAG: Pattern/ExprSpecific:               {#yArg: Foo...#}[#Foo#];
+    // COMPLETE_AFTER_VARARG_WITH_PREV_PARAM-DAG: Pattern/ExprSpecific:               {#zArg: Foo...#}[#Foo#];
+    // COMPLETE_AFTER_VARARG_WITH_PREV_PARAM: End completions
```
These tests don't look right. For `xArg: #^HERE^#`, expected suggestion are global expression completions.
Also, `xArg: #^COMPLETE_AFTER_VARARG_WITH_PREV_PARAM^#` is currently the same as `COMPLETE_AFTER_VARARG `. This probably meant `xArg: .bar, #^HERE^#` which expects argument labels _and_ global expressions.

ahoppen replied:
Of course 🤦  Updated the tests to test the right thing and the PR to behave according to the updated tests.

ahoppen committed:
[CodeComplete] Complete argument labels after vararg

Currently, if the code completion token is after a label that refers to a vararg, it is part of that VarargExpansionExpr, so we don’t suggest the subsequent parameter’s label.

Add special handling to detect this situation and suggest the parameter label.

Fixes rdar://76355192

ahoppen commented:
@swift-ci Please test

ahoppen commented:
@swift-ci Please smoke test macOS

rintaro reviewed (approved):
Thanks!
